flink-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/SimpleConsumerThread.java

```diff
@@ -381,6 +381,10 @@ else if (partitionsRemoved) {
 								partitionsIterator.remove();
 								continue partitionsLoop;
 							}
+
+							if (value == null) {
+								continue;
+							}
```

Comment:
Would it make sense to do the `null` checking inside `emitRecord(...)`?
Otherwise, we wouldn't be updating the state for skipped records, and therefore not accounting it as "already processed".

I don't think it really matters, since we aren't outputting anything anyway, but I see at least one minor advantage that might deserve changing it: If we fail during a series of continuous skipped records, we won't be wasting any overhead re-processing them on restore.